base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -86,6 +86,14 @@ pub trait BlockchainBackend: Send + Sync {
     fn fetch_mmr_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
     /// Returns only the MMR merkle root without the state of the roaring bitmap.
     fn fetch_mmr_only_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
+    /// Fetches the merklish root for the MMR tree identified by the key after the current additions and deletions have
+    /// temporarily been applied. Deletions of hashes from the MMR can only be applied for UTXOs.
+    fn fetch_future_mmr_root(
```

Comment:
We're not really "fetching" anything here. 
Is there a name that more accurately describes the function of this method?

`calculate_mmr_root` perhaps? The parameters are a strong indicator that it's not the current root we'll be getting.
`calculate_mmr_root_after_changes` too wordy?

Reply:
I agree the parameters provide a good indicator of its function, the function name was changed to `calculate_mmr_root`.